botocore/args.py

```diff
@@ -900,8 +900,20 @@ def _handle_checksum_config(
                 config_value=value,
                 valid_options=valid_options,
             )
+        self._register_checksum_config_feature_ids(value, config_key)
         config_kwargs[config_key] = value
 
+    def _register_checksum_config_feature_ids(self, value, config_key):
+        if config_key == "request_checksum_calculation":
+            checksum_calculation_feature_id = (
```

Comment:
The `checksum_calculation_feature_id` name only makes sense in the if statement and not in the elif part. We should use something like `checksum_config_feature_id`.

Also, we should handle the case where `config_key` is not `"request_checksum_calculation"` or `"response_checksum_validation"`. We shouldn't fail for a user-agent related issue. I think this would currently fail with:
```
NameError: name 'checksum_calculation_feature_id' is not defined
```